spec/unit/shoulda/matchers/active_record/validate_uniqueness_of_matcher_spec.rb

```diff
@@ -1043,6 +1043,53 @@
     end
   end
 
+  context 'when the writer method for the attribute changes incoming values' do
+    context 'and the matcher knows nothing of this' do
+      it 'raises a CouldNotSetAttributeError' do
+        model = define_model_validating_uniqueness(
+          attribute_name: :name,
+          validation_options: { case_sensitive: false }
+        )
+
+        model.class_eval do
+          def name=(name)
+            super(name.upcase)
+          end
+        end
+
+        assertion = lambda do
+          expect(model.new).
+            to validate_uniqueness_of(:name).
+            case_insensitive
+        end
+
+        expect(&assertion).to raise_error(
+          Shoulda::Matchers::ActiveModel::AllowValueMatcher::CouldNotSetAttributeError
+        )
+      end
+    end
+
+    context 'and the matcher knows how given values get changed' do
+      it 'accepts (and not raise an error)' do
+        model = define_model_validating_uniqueness(
+          attribute_name: :name,
+          validation_options: { case_sensitive: false }
+        )
+
+        model.class_eval do
+          def name=(name)
+            super(name.upcase)
+          end
+        end
+
+        expect(model.new).
+          to validate_uniqueness_of(:name).
+          case_insensitive.
+          converting_values("a" => "A")
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.